ompi/tools/mpisync/mpigclock.c

```diff
@@ -60,28 +60,63 @@ static double mpigclock_measure_offset_adaptive(MPI_Comm comm, int root, int pee
  */
 double mpigclock_sync_linear(MPI_Comm comm, int root, double *rtt)
 {
-    int i, rank, commsize;
+    int peer, rank, commsize;
     double ret = 0;
 
     MPI_Comm_rank(comm, &rank);
     MPI_Comm_size(comm, &commsize);
 
-    if (commsize < 2) {
-        *rtt = 0.0;
-        return 0.0;
-    }
-
-    for (i = 1; i < commsize; i++) {
+    for (peer = 1; peer < commsize; peer++) {
         MPI_Barrier(comm);
-        if (rank == root || rank == i) {
-            ret = mpigclock_measure_offset_adaptive(comm, root, i, rtt);
+        if (rank == root || rank == peer) {
+            ret = mpigclock_measure_offset_adaptive(comm, root, peer, rtt, 0.0);
         }
     }
     return ret;
 }
 
+/*
+ * mpigclock_sync_log: Clock synchronization algorithm with O(logn) steps.
+ * rtt argumrnt does not have a meaning
+ */
```

Comment:
I would keep rtt equal to the value measured for the parent. This would still provide precision estimation (though with less level of confidence than linear)